File: codes/omp_hello.cpp
#include <stdio.h>
#include <omp.h>

int main()
{
  int nthreads, tid;

  #pragma omp parallel private(nthreads, tid)
  {
    tid = omp_get_thread_num();
    // C++ cout is not thread-safe
    printf("Hello, world! I am thread %d\n", tid);

    #pragma omp barrier
    if (tid == 0) 
    {
      nthreads = omp_get_num_threads();
      printf("Number of threads = %d\n", nthreads);
    }
  }

  return 0;
}

File: codes/omp_hello.f
      program main
      implicit none
      include "omp_lib.h"
      integer nthreads, tid

C$OMP PARALLEL PRIVATE(nthreads, tid)
      tid = OMP_GET_THREAD_NUM()
      print *, "Hello, world! I am thread ", tid
C$OMP BARRIER
      if (tid .eq. 0) then
        nthreads = OMP_GET_NUM_THREADS()
        print *, "Number of threads = ", nthreads
      end if
C$OMP END PARALLEL

      stop
      end
